include/swift/AST/DiagnosticsSema.def

```diff
@@ -1109,32 +1109,34 @@ ERROR(static_requires_initializer,none,
       "expression or getter/setter specifier", (StaticSpellingKind))
 ERROR(pattern_type_access,none,
       "%select{%select{variable|constant}0|property}1 "
-      "%select{must be declared %select{private|fileprivate|internal|PUBLIC}4"
+      "%select{must be declared %select{"
+      "%select{private|fileprivate|internal|PUBLIC}3|private or fileprivate}5"
       "|cannot be declared %select{PRIVATE|fileprivate|internal|public}3}2 "
       "because its type uses "
       "%select{a private|a fileprivate|an internal|PUBLIC}4 type",
-      (bool, bool, bool, Accessibility, Accessibility))
+      (bool, bool, bool, Accessibility, Accessibility, bool))
```

Comment:
Just seeing this now: please put the new flag directly after the access level it affects (for all of the modified diagnostics). The jump from argument 3 to argument 5 confused me for a bit.